lib/collection/response.js

```diff
@@ -142,17 +142,24 @@ var util = require('../util'),
     supportsBuffer = (typeof Buffer !== undefined) && _.isFunction(Buffer.byteLength),
 
     /**
-     * Normalises an input Buffer or buffer.toJSON() into a Buffer or ArrayBuffer.
+     * Normalizes an input Buffer, Buffer.toJSON() or base64 string into a Buffer or ArrayBuffer.
      *
      * @private
-     * @param {Buffer|Object} stream - An instance of Buffer of an object representation of Buffer(Buffer.toJSON())
+     * @param {Buffer|Object|String} stream - An instance of Buffer, Buffer.toJSON(), or Buffer.toString('base64')
      * @returns {Buffer|ArrayBuffer|undefined}
      */
-    normaliseStream = function (stream) {
+    normalizeStream = function (stream) {
+        // create buffer from buffer's JSON representation
         if (stream && stream.type === BUFFER && _.isArray(stream.data)) {
             // @todo Add tests for Browser environments, where ArrayBuffer is returned instead of Buffer
-            return typeof Buffer === FUNCTION ? new Buffer(stream.data) : new Uint8Array(stream.data).buffer;
+            return typeof Buffer === FUNCTION ? Buffer.from(stream.data) : new Uint8Array(stream.data).buffer;
         }
+        // create buffer from base64 string
+        else if (typeof stream === 'string') {
```

Comment:
Can you add comment here stating that based on the `if` statement condition structure, it is imperative that the string check lies in the ease block, otherwise blank string would be erroneously not converted to buffer. 

- [ ] add test for blank string stream check.